### PR TITLE
SDI-477: ✨ Upgrade to spring boot 3

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: hmpps-sqs-spring-boot-autoconfigure/src/main/kotlin/uk/gov/justice/hmpps/sqs/HmppsSqsConfiguration.kt [hunk 1]
```diff
@@ -17,7 +18,7 @@ import org.springframework.context.annotation.Import
 
 @Configuration
 @EnableConfigurationProperties(HmppsSqsProperties::class)
-@AutoConfigureBefore(WebFluxAutoConfiguration::class)
+@AutoConfigureBefore(WebFluxAutoConfiguration::class, WebMvcAutoConfiguration::class)
```
Comment (petergphillips) [inline hmpps-sqs-spring-boot-autoconfigure/src/main/kotlin/uk/gov/justice/hmpps/sqs/HmppsSqsConfiguration.kt:21]: ensure that we configure before the mvc version too

File: hmpps-sqs-spring-boot-autoconfigure/src/main/kotlin/uk/gov/justice/hmpps/sqs/HmppsSqsProperties.kt [hunk 2]
```diff
@@ -1,9 +1,7 @@
 package uk.gov.justice.hmpps.sqs
 
 import org.springframework.boot.context.properties.ConfigurationProperties
-import org.springframework.boot.context.properties.ConstructorBinding
 
-@ConstructorBinding
```
Comment (petergphillips) [inline hmpps-sqs-spring-boot-autoconfigure/src/main/kotlin/uk/gov/justice/hmpps/sqs/HmppsSqsProperties.kt:6]: not required for spring boot 3

File: test-app-reactive/build.gradle.kts [hunk 3]
```diff
@@ -17,21 +17,19 @@ dependencies {
   implementation("org.jetbrains.kotlinx:kotlinx-coroutines-reactor")
   implementation("org.jetbrains.kotlinx:kotlinx-coroutines-jdk8")
 
-  implementation("org.springdoc:springdoc-openapi-webflux-ui:1.6.13")
-  implementation("org.springdoc:springdoc-openapi-kotlin:1.6.13")
-  implementation("org.springdoc:springdoc-openapi-security:1.6.13")
+  implementation("org.springdoc:springdoc-openapi-starter-webflux-ui:2.0.0")
```
Comment (petergphillips) [inline test-app-reactive/build.gradle.kts:20]: note the new location

**Review by andymarke — APPROVED** [review]
LGTM

**Pull request merged**